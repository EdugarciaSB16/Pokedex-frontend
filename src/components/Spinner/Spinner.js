/**
 * Componenten spinner para el estado de carga de la aplicaciÃ³n
 * 
 */
// Dependencias
import React from 'react';

export const Spinner = () => {
    return <div className="pokemon"></div>;
};
