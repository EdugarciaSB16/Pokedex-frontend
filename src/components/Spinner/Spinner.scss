.pokemon {
    height: 100px;
    width: 100px;
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    top: 50%;
    text-align: center;
    background: linear-gradient(to bottom, rgb(254, 0, 1) 50%, white 50%);
    border-radius: 50%;
    border: 8px solid black;
    animation: spin 1s linear infinite;

    @keyframes spin {
        from {
            transform: rotate(0deg);
        }
        to {
            transform: rotate(360deg);
        }
    }
}
