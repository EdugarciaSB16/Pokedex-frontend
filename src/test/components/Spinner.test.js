/**
 * Test al componente spinner
 * 
 */
// Dependencias
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
// Componentes
import { Spinner } from '../../components/Spinner/Spinner';

configure({ adapter: new Adapter() });

describe('Pruebas en el componente <Spinner />', () => {
    test('Debe de renderizar el componente correctamente', () => {
        const wrapper = shallow(<Spinner />);
        expect(wrapper).toMatchSnapshot();
    });
});