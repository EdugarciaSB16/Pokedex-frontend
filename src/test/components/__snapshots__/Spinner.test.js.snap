// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Spinner /> Debe de renderizar el componente correctamente 1`] = `
<div
  className="pokemon"
/>
`;
